drivers/soundwire/master.c

```diff
@@ -102,66 +99,60 @@ EXPORT_SYMBOL_GPL(sdw_master_device_add);
  * This function is the dual of sdw_master_device_add(), itreleases
  * all link-specific resources and unregisters the device.
  */
-int sdw_master_device_del(struct sdw_master_device *md)
+int sdw_master_device_del(struct sdw_bus *bus)
 {
 	int ret = 0;
 
-	if (md && md->link_ops && md->link_ops->del) {
-		ret = md->link_ops->del(md);
+	if (bus->link_ops && bus->link_ops->del) {
+		ret = bus->link_ops->del(bus);
 		if (ret < 0) {
-			dev_err(&md->dev, "link_ops del callback failed: %d\n",
+			dev_err(bus->dev, "link_ops del callback failed: %d\n",
 				ret);
 			return ret;
 		}
 	}
 
-	device_unregister(&md->dev);
+	pm_runtime_disable(&bus->md->dev);
```

Comment:
you dont enable runtime PM in add? its not symmetrical

Reply:
@ranj063 I think we can't enable pm_runtime before its parent is enabled. So I call pm_runtime_time in intel_master_startup(). But you are right. We should do it symmetrically. So I remove pm_runtime_disable() to sdw_intel_cleanup()